frame/democracy/src/lib.rs

```diff
@@ -688,6 +718,62 @@ decl_module! {
 			Self::internal_cancel_referendum(ref_index);
 		}
 
+		/// Permanently place a proposal into the blacklist. This prevents it from ever being
+		/// proposed again.
+		///
+		/// If called on an queued public or external proposal, then this will result in it being
+		/// removed. If the `ref_index` supplied is an active referendum with the proposal hash,
+		/// then it will be cancelled.
+		///
+		/// The dispatch origin of this call must be `BlacklistOrigin`.
+		///
+		/// - `proposal_hash`: The proposal hash to blacklist permanently.
+		/// - `ref_index`: An ongoing referendum whose hash is `proposal_hash`, which will be
+		/// cancelled.
+		#[weight = (T::WeightInfo::blacklist(*prop_count), DispatchClass::Operational)]
+		fn blacklist(origin,
+			proposal_hash: T::Hash,
+			maybe_ref_index: Option<ReferendumIndex>,
+			#[compact] prop_count: u32,
+		) {
+			T::BlacklistOrigin::ensure_origin(origin)?;
+
+			let real_prop_count = PublicProps::<T>::decode_len().unwrap_or(0) as u32;
+			ensure!(real_prop_count <= prop_count, Error::<T>::InvalidWitness);
+
+			// Insert the proposal into the blacklist.
+			let permanent = (T::BlockNumber::max_value(), Vec::<T::AccountId>::new());
+			Blacklist::<T>::insert(&proposal_hash, permanent);
+
+			// Remove the queued proposal, if it's there.
+			PublicProps::<T>::mutate(|props| {
```

Comment:
this operation also reads and writes PublicProps which is `O(p)`, it seems the benchmark miss it.

Reply:
ah this one has been fixed